nntrainer/utils/ini_wrapper.h

```diff
@@ -0,0 +1,178 @@
+   * @param section_name section name to override, if empty, section name is not
+   * updated
+   * @param entry_str entry string to override the given section name
+   */
+  IniSection(IniSection &from, const std::string &section_name,
+             const std::string &entry_str);
+
+  /**
+   * @brief Construct a new Ini Section object
+   *
+   * @param from Ini Section to copy from
+   * @param entry_str entry string to override the given section name
+   */
+  IniSection(IniSection &from, const std::string &entry_str) :
+    IniSection(from, "", entry_str) {}
+
+  IniSection() = default;
+  ~IniSection() = default;
+
+  /**
+   * @brief +=operator from IniSection
+   *
+   * @param rhs operand to add
+   * @return IniSection& this
+   */
+  IniSection &operator+=(const IniSection &rhs) {
+    setEntry(rhs.entry);
+    return *this;
+  }
+
+  /**
+   * @brief + operator from IniSection
+   *
+   * @param rhs operand to add
+   * @return IniSection new Inisection
+   */
+  IniSection operator+(const IniSection &rhs) const {
+    return IniSection(*this) += rhs;
+  }
+
+  /**
+   * @brief += operator from string
+   *
+   * @param s string representation to add
+   * @return IniSection& this
+   */
+  IniSection &operator+=(const std::string &s) {
+    setEntry(s);
+    return *this;
+  }
+
+  /**
+   * @brief + operator from string
+   *
+   * @param s string representation to add
+   * @return IniSection Newly created section
+   */
+  IniSection operator+(const std::string &s) { return IniSection(*this) += s; }
+
+  /**
+   * @brief equal operator between ini section
+   *
+   * @param rhs ini section to compare
+   * @return true two inisections are equal
+   * @return false two ini sections are not equal
+   */
+  bool operator==(const IniSection &rhs) const {
+    return section_name == rhs.section_name && entry == rhs.entry;
+  }
+
+  /**
+   * @brief  not equal operator between ini section
+   *
+   * @param rhs ini section to compare
+   * @return true two inisections are not equal
+   * @return false two inisections are equal
+   */
+  bool operator!=(const IniSection &rhs) const { return !operator==(rhs); }
+
+  /**
+   * @brief print out a section
+   *
+   * @param out ostream to print
+   */
+  void print(std::ostream &out);
+
+  /**
+   * @brief Get the Name object
+   *
+   * @return std::string section name
+   */
+  std::string getName() { return section_name; }
+
+private:
+  /**
+   * @brief Set the Entry
+   *
+   * @param entry set entry from a given map
+   */
+  void setEntry(const std::map<std::string, std::string> &entry);
+
+  /**
+   * @brief set entry from the string representation
+   *
+   * @param entry_str setEntry as "Type = neuralnetwork | decayrate = 0.96 |
+   * -epochs = 1" will delete epochs, and overwrite type and decayrate
+   */
+  void setEntry(const std::string &entry_str);
+
+  std::string section_name; /**< section name of the ini section */
+  std::map<std::string, std::string>
```

Comment:
consider using `unordered_map` to make finding a section faster.

Reply:
I changed to ordered map to keep the order predictable, I checked with current tests and overhead is minimal.

Reply:
> I changed to ordered map to keep the order predictable,

Any reason for this? For the ini input file, we can't assume any order.

Reply:
This is for testing purpose. If this is critical to the performance, I could change but I think it's fine as one section wouldn't have more than 10 or 20 entries.

Reply:
Done